File: TiaCodegen/Commands/Coils/BaseNPCoil.cs
using TiaCodegen.Commands.Signals;
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Coils
{
    public class BaseNPCoil : BaseCoil
    {
        public BaseNPCoil(Signal signal, Signal helpSignal, IOperationOrSignal op = null) : base(signal, op)
        {
            HelpSignal = helpSignal;
            if (HelpSignal != null)
                Children.Add(helpSignal);
        }

        public Signal HelpSignal { get; set; }
    }
}

File: TiaCodegen-ts/src/Commands/Coils/BaseNPCoil.ts
import { IOperationOrSignal } from '../../Interfaces/IOperationOrSignal.js';
import { Signal } from '../Signals/Signal.js';
import { BaseCoil } from './BaseCoil.js';

export class BaseNPCoil extends BaseCoil {
    helpSignal: Signal | null;

    constructor(signal: Signal, helpSignal: Signal | null, op: IOperationOrSignal | null = null) {
        super(signal, op);
        this.helpSignal = helpSignal;
        if (helpSignal !== null) {
            this.children.push(helpSignal);
        }
    }
}
